### PR TITLE
ci: Further tweaks to building NSS

### PR DESCRIPTION
**Pull request by larseggert**
Picks up the comments @martinthomson left on #1710

### CODE REVIEW

File: .github/actions/nss/action.yml [hunk 1]
```diff
@@ -1,19 +1,53 @@
 name: Fetch and build NSS
 
+inputs:
+  type:
+    description: 'Whether to do a debug or release build of NSS'
+    required: true
+    default: 'Debug'
```
Comment (martinthomson) [inline .github/actions/nss/action.yml:7]: Having both required and default seem to be incompatible.

I would also recommend a default of 'Release' on the basis that we won't be debugging NSS code in CI, so the extra performance is probably worth it.

File: .github/actions/nss/action.yml [hunk 2]
```diff
@@ -1,19 +1,53 @@
...
+    - name: Build
       shell: bash
       run: |
-        git clone --depth=1 https://github.com/nss-dev/nspr "$NSPR_DIR"
-        git clone --depth=1 https://github.com/nss-dev/nss "$NSS_DIR"
-        $NSS_DIR/build.sh -g -Ddisable_tests=1 -o
+        # We want to do an optimized build for accurate CPU profiling, but
+        # we also want debug symbols for that, which the normal optimized NSS
+        # build process doesn't provide.
+        [ "${{ runner.os }}" != "Windows" ] && export CFLAGS=-ggdb3
```
Comment (martinthomson) [inline .github/actions/nss/action.yml:37]: Does this belong inside the if statement then?

File: .github/actions/nss/action.yml [hunk 3]
```diff
@@ -1,19 +1,53 @@
...
+        if [ "${{ inputs.type }}" != "Debug" ]; then
+          export OPT="-o"
+          export NSS_TARGET=Release
+        else
+          export NSS_TARGET=Debug
+        fi
+        $NSS_DIR/build.sh -g -Ddisable_tests=1 $OPT --static
+        echo "NSS_TARGET=$NSS_TARGET" >> "$GITHUB_ENV"
+        export NSS_OUT="$NSS_DIR/../dist/$NSS_TARGET"
```
Comment (martinthomson) [inline .github/actions/nss/action.yml:46]:
```suggestion
        NSS_OUT="$NSS_DIR/../dist/$NSS_TARGET"
```

I'm fairly sure that you don't need `export` if you are using the string in this shell.

File: neqo-crypto/build.rs [hunk 4]
```diff
@@ -147,10 +148,9 @@ fn build_nss(dir: PathBuf) {
     let mut build_nss = vec![
         String::from("./build.sh"),
         String::from("-Ddisable_tests=1"),
+        String::from("--static"),
```
Comment (martinthomson) [inline neqo-crypto/build.rs:151]: I don't understand this change.  Can you explain?

File: .github/workflows/bench.yml [hunk 5]
```diff
@@ -8,8 +8,8 @@ env:
   CARGO_TERM_COLOR: always
   RUST_BACKTRACE: 1
   TOOLCHAIN: nightly
-  RUSTFLAGS: -C link-arg=-fuse-ld=lld -C link-arg=-Wl,--no-rosegment
-  PERF_CMD: record -o perf.data -F997 --call-graph lbr -g
+  RUSTFLAGS: -C link-arg=-fuse-ld=lld -C link-arg=-Wl,--no-rosegment, -Cforce-frame-pointers=yes
+  PERF_CMD: record -o perf.data -F997 --call-graph fp -g
```
Comment (martinthomson) [inline .github/workflows/bench.yml:12]: Unrelated?

File: neqo-crypto/build.rs (hunk 4 above)
Comment (larseggert) [reply neqo-crypto/build.rs:151]: I had queued up a change with a comment that I forgot to push. 
```
// Generate static libraries in addition to shared libraries.
```
Basically, `--static` seems to not only build static libraries, it seems to build static libraries **in addition** to dynamic ones. So no reason to not always use it.

File: .github/workflows/bench.yml (hunk 5 above)
Comment (larseggert) [reply .github/workflows/bench.yml:12]: Yes. Still mucking around with trying to find the best flags for `perf`, and don't want to do a whole other parallel CI run for that.

**Review by martinthomson — APPROVED** [review]
